src/System.Private.ServiceModel/src/Extensions/XmlReaderExtensions.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Xml;
+
+namespace System.ServiceModel
+{
+	internal static class XmlReaderExtensions
+	{
+	    internal static string ReadElementString(this XmlReader reader)
+	    {
+            if (reader.MoveToContent() != XmlNodeType.Element)
+            {
+                var lineInfo = reader as IXmlLineInfo;
+                throw new XmlException(SR.Format(SR.Xml_InvalidNodeType, reader.NodeType.ToString()), null, lineInfo?.LineNumber ?? 0, lineInfo?.LinePosition ?? 0);
+            }
+
+            return reader.ReadElementContentAsString();
+        }
+
+	    internal static string ReadElementString(this XmlReader reader, string localname, string ns)
+	    {
+            if (reader.MoveToContent() != XmlNodeType.Element)
+            {
+                var lineInfo = reader as IXmlLineInfo;
+                throw new XmlException(SR.Format(SR.Xml_InvalidNodeType, reader.NodeType.ToString()), null, lineInfo?.LineNumber ?? 0, lineInfo?.LinePosition ?? 0);
+            }
+
+	        return reader.ReadElementContentAsString(localname, ns);
+	    }
+	}
+}
```

Comment:
Not quite a copy, but equivalent functionality. It can't be directly copied as it's a virtual function. By calling reader.ReadElementContentAsString, any derived classes have their correct implementation used.